Dart/src/com/jetbrains/lang/dart/ide/errorTreeView/DartProblem.java

```diff
@@ -173,21 +174,24 @@ private void ensureInitialized() {
                                                          @Nullable @Nls String correction,
                                                          @Nullable @NonNls String url) {
     // First include the error message.
-    HtmlBuilder htmlBuilder = new HtmlBuilder().append(message);
+    HtmlBuilder htmlBuilder = new HtmlBuilder().append(message).append(HtmlChunk.br());
 
     // For each context message, include the message and location.
     if (contextMessages != null && !contextMessages.isEmpty()) {
+      List<HtmlChunk> liChunks = new ArrayList<>(contextMessages.size() * 2);
       for (DiagnosticMessage contextMessage : contextMessages) {
         Location location = contextMessage.getLocation();
         @NlsSafe String msg = StringUtil.trimEnd(contextMessage.getMessage(), '.');
         String locationText = PathUtil.getFileName(location.getFile()) + ":" + location.getStartLine();
-        htmlBuilder.append(HtmlChunk.br()).append(HtmlChunk.br()).append(msg).append(" (" + locationText + ").");
+        liChunks.add(HtmlChunk.li());
+        liChunks.add(HtmlChunk.text(msg + " (" + locationText + ")."));
```

Comment:
I guess it should be `HtmlChunk.li().addText(msg + " (" + locationText + ").")` so that the text is surrounded with `<li>` and `</li>`.